src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/SessionService.kt

```diff
@@ -53,30 +54,42 @@ class SessionService(
     val requestedBookableStartDate = today.plusDays(noticeDaysMin ?: policyNoticeDaysMin)
     val requestedBookableEndDate = today.plusDays(noticeDaysMax ?: policyNoticeDaysMax)
 
+    // ensure the prisoner - if supplied belongs to the same prison as supplied prisonCode
+    prisonerId?.let {
+      prisonerValidationService.validatePrisonerIsFromPrison(prisonerId, prisonCode)
+    }
```

Comment:
Not really for this review but why is prisonerId optional...

Reply:
Yes, need to pick this up as tech debt. We could also remove prisonId parameter and only use prisonerId and get the prisoner's prison ID from a Prison API call so that we always get the sessions for the right prison if the front end does not use only prisonId option.